libminifi/include/utils/HTTPClient.h

```diff
@@ -300,6 +319,9 @@ class BaseHTTPClient {
   virtual void setUploadCallback(HTTPUploadCallback *callbackObj) {
   }
 
+  virtual void setSeekFunction(HTTPUploadCallback *callbackObj) {
+  }
```

Comment:
it seems like `HTTPClient` is the only class inheriting from it, and noone directly instantiates a BaseHTTPClient, could we make these noop virtual methods pure virtual?

Reply:
Done in [fd3a7c6](https://github.com/apache/nifi-minifi-cpp/pull/897/commits/fd3a7c653202c7cdd460153175d3deb13b7651ce)